### PR TITLE
HYD-4746 Add error support to jasmine-n-matchers 2.x

### PR DESCRIPTION
**Pull request by wayneseymour**
We need to internalize jasmine-n-matchers from the public npm/github
repos.
We also need to add tests to it, and support for errors.

Upload the module to our repos.
Master branch should support jasmine 2.x.

 - Added an custom equality tester, for errors.
 - Added tests.

Signed-off-by: Wayne Kenneth Seymour III <wayne.k.seymour.iii@intel.com>

### CODE REVIEW

File: jasmine-n-matchers.js [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
 beforeEach(function () {
+  'use strict';
+
+  var pp = jasmine.pp.bind(jasmine);
+
+  jasmine.addCustomEqualityTester(errorEquality);
+
+  function errorEquality (a, b) {
+    if (a instanceof Error && b instanceof Error)
+      return a.message === b.message;
```
Comment (jgrund) [inline jasmine-n-matchers.js]:
```
  var constructorLens = fp.lensProp('constructor');

  var eqConstructor = fp.eqFn(constructorLens, constructorLens);
```

Comment (jgrund) [inline jasmine-n-matchers.js]: Also you should check that both are instances of `Error`.

File: jasmine-n-matchers.js [hunk 2]
```diff
@@ -79,19 +93,28 @@ beforeEach(function () {
...
+          var id = actual.and.identity();
 
-          if (result.pass)
-            result.message = 'Expected spy ' + actualIdentity + ' not to have been called with ' +
-              jasmine.pp(expectedArgs) + ' ' + n + ' time(s) but it was.';
-          else
-            result.message = 'Expected spy ' + actualIdentity + ' to have been found with ' +
-              jasmine.pp(expectedArgs) + ' ' + n + ' time(s) but it was found ' + foundCount + ' time(s).\n\n' +
-              'Spy '+ actual.and.identity() + ' call listing:\n' + jasmine.pp(allActualArgs) + '.';
+          result.message = buildMessage();
```
Comment (jgrund) [inline jasmine-n-matchers.js]: You usually don't want to make a function unless you are reusing the code somewhere. Otherwise, it's just a goto.

Comment (wayneseymour) [inline jasmine-n-matchers.js]: Visual Basic

File: jasmine-n-matchers.js (hunk 1 above)
Comment (wayneseymour) [inline jasmine-n-matchers.js]: @johnsonw agrees too.  Adding now

File: test/jasmine-n-matchers-test.js [hunk 3]
```diff
@@ -161,24 +172,114 @@ describe('jasmine n matchers', function () {
...
+          });
+
+          it('should pass against a URIError expectation', function () {
+            expect(errSpy).toHaveBeenCalledOnceWith(new URIError('whoops'));
+          });
+
+        });
       });
     });
+    //describe('toHaveBeenCalledOnceWith', function () {
```
Comment (jgrund) [inline test/jasmine-n-matchers-test.js]: dead code.

File: jasmine-n-matchers.js (hunk 1 above)
Comment (jgrund) [inline jasmine-n-matchers.js]: `Object.getPrototypeOf(a).constructor === Object.getPrototypeOf(b).constructor`

**Comment by johnsonw** [comment]
:+1:

**Comment by wayneseymour** [comment]
Sure thing.

**Comment by jgrund** [comment]
:+1:

**Comment by johnsonw** [comment]
:+1:

**Comment by jgrund** [comment]
Merged via: https://github.com/intel-js/jasmine-n-matchers/commit/6b683e79f4cff0eef481565fef1117e0b0a2927b

**Pull request closed without merging**